packages/web-app-files/src/services/folder/loaderSpace.ts

```diff
@@ -53,9 +62,17 @@ export class FolderLoaderSpace implements FolderLoader {
       try {
         store.commit('Files/CLEAR_CURRENT_FILES_LIST')
 
-        const resources = yield webdav.listFiles(space, { path, fileId })
-        let currentFolder = resources.shift()
-        replaceInvalidFileRoute({ space, resource: currentFolder, path, fileId })
+        const resourcesPromise = webdav.listFiles(space, { path, fileId })
```

Comment:
This doesn't work - the `listFiles` call has an internal path correction mechanism. Works as follows:
1. you reach a URL with path and fileId (please note: CERNbox is not supposed to use the fileIds in URLs. We build a disable switch just for you, see documentation for `options.routing.idBased` in https://owncloud.dev/clients/web/getting-started/#options
2. both the path and the fileId get passed into `listFiles` here
3. the request is done by path (webdav works path based, so that's only natural)
4. the fileId from the webdav response of (3) is compared with the fileId provided in the params
5. if the fileIds don't match, we request "the new path" by fileId, using the `meta` endpoint. This can happen if you open a bookmarked URL and the relative item path has changed since you bookmarked the URL (e.g. a folder has been renamed or moved to a different location)
6. the `listFiles` request happens again, but with the corrected path

IMPORTANT: if the fileId changed (due to e.g. rename) it is possible that a new file/folder has been created in the old, bookmarked location / path. You would request shares for a completely different file/folder in that case.

This is the reason why we have to await the propfind before we request the shares. Not possible to parallelize this!

I see one option though: we could
1. await/yield a `getFileInfo` request, which is a propfind with depth 0 internally. This would solve our path correction issue
2. we use your approach to parallelize a `listFiles` (= propfind with depth 1) and the OCS getShares call.

I can't tell if it's worth the effort. My assumption is, that a propfind with depth 0 is reasonably faster than a propfind with depth 1 for CERNbox, but you need to tell me if that assumption is valid.
If you want to I can provide a pull request solving the stuff above - or if your time permits you could solve this as well of course.
In any case I'd be very happy to see (and merge) a pull request with your other changes!! Really good that you started parallelizing requests. 👍

Reply:
What if I use the switch `routing.idBased` in here? If this is false, then we can safely assume that the provided id is the expected one, right? Or at least more or less (see last paragraph)...

But actually, I think that this happens already (did this some time ago so I'm not sure anymore).. Basically, if I land on a page the `fileId` will be undefined (because we have `idBased = false`) so the url doesn't contain it. So we will retrieve it just like the previous behaviour, i.e it will be the first thing we do. The id is only defined if this method is reached having before PROPFINDING the current folder. 
But, ofc, if `idBased = true`, we will have always an `id` to pass to this method (unless we distinguish between an id given by the route or by the current folder info). So, if we separated the id the was provided by the route from the one provided by the current folder PROPFIND, we would be able to force the re-check only in the first case.

The only issue I see is if something changes in the background (some other open tab, for example), so our current folder info is no longer up to date. Not sure we would want to accept this corner case inconsistency.

Reply:
@kulmann ^

Reply:
Just checked, and since we eventually `replaceInvalidFileRoute` and redirect the user to the new path, what happens is that what is presented is the correct information. The worst that happens is that there is an extra call to the shares for the wrong folder that is not the current one.
isn't it better to do a bad, unnecessary, request on the (exceptional) possibility that a path has changed, if by doing so we are improving the more common case that the path is correct?